### PR TITLE
Improved the setup script

### PR DESCRIPTION
**Pull request by pboothe**
The setup script now will add all the Google netblocks to the whitelist of netblocks.  Other pull requests will follow, each adding these netblocks to the slices that have specialized rsync configurations.

### CODE REVIEW

File: slicebase/init/setup_rsync.sh [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
+#   nslookup -q=TXT _cloud-netblocks.googleusercontent.com  8.8.8.8 \
+#    | grep text \
+#    | sed -e 's/.*=spf1 //' -e 's/?all.*//' -e 's/include://g' -e 's/ /\n/g'  \
+#    | while read; do nslookup -q=TXT $REPLY 8.8.8.8; done \
+#    | grep 'text = ' \
+#    | sed -e 's/.*spf1 //' -e 's/ ?all.*//' -e 's/ /\n/g' \
+#    | grep ^ip4 \
+#    | sed -e 's/ip4://' \
+#    | xargs \
+#    | sed -e 's/ /, /g'
```
Comment (salarcon215) [inline slicebase/init/setup_rsync.sh]: This is great, putting the nslookup query right in the script. That leads me to ask, though, how static are these IP assignments? Does it make more sense to use this lookup right in the script rather that to declare the IPs themselves?

Comment (pboothe) [inline slicebase/init/setup_rsync.sh]: I don't know how dynamic the list is. I suspect not very?

But I get a little twinge when I think of making rsync setup dependent on nslookup to 8.8.8.8 succeeding in just the right way. That's why I put the script in comments and the netblocks in code.

Comment (salarcon215) [inline slicebase/init/setup_rsync.sh]: I had that thought as soon as I hit "comment". So, friendly push-back part dieux: What about putting the nslookup in a time-bounded if; if the if succeeds, update the static list; if the if fails, rely on the previously-derived static list?

Comment (salarcon215) [inline slicebase/init/setup_rsync.sh]: Another way to ask is, if the IPs change, how will we find out about it?

Comment (salarcon215) [inline slicebase/init/setup_rsync.sh]: Decision: LGTM.
pboothe responded: 

> It is reasonable to have a fallback if the script is run regularly. If the process is run once, ever, like this is, then having a fallback is a bad idea - it leaves the slice permanently in "fallback mode" or permanently in "live data mode".
> 
> This setup should work or it should fail and cause everything else to fail. What I don't know (and I suspect nobody knows without trying it) is whether nslookup is definitely installed at the point that this script is run, and whether ops is comfortable requiring that every node have reachability to 8.8.8.8 when this script is run.
> 
> We will know if the static list doesn't work if we start trying to scrape from GCE instances with IPs that are not in the list and they get permission denied. I don't anticipate the lifetime of the rsync pipeline and this mode of slice setup will be long enough for this to be a problem.
> 
> Taking on new dependencies (or not) is your choice as the person doing operations. But were our positions reversed I would want the simplicity of the static list. If you still want me to make it dynamically generated, then I will.

Salarcon responded:

> 
> Ok, I'm satisfied with this based on
> 1) infrequency of the run
> 2) occurrence early in the hot setup process
> 4) short intended lifetime of the script
> 5) assurance that a problem would be detected
> 6) the fact that rsync errors are back
> 
> So go forth and merge. I'll add these notes to the PR.

**Pull request merged**